test/compilable/test17807.d

```diff
@@ -0,0 +1,17 @@
+// REQUIRED_ARGS: -o- -w
+
+int bug17807(){
+    int y=0;
+    Lswitch: switch(2){
+        { case 0: break; }
+        enum x=0;
+        struct S{ enum x=0; }
+        int foo(){
+            return 0;
+        }
+        default: y=x+S.x+foo();
+        static foreach(i;1..5)
```

Comment:
Interestingly, if you remove `static` here, there's a ICE in the backend.  Has a bug report been raised for this?

Reply:
Nice catch. I couldn't find an existing issue. I have opened a new issue here: https://issues.dlang.org/show_bug.cgi?id=17834
(However, this bug will disappear once the deprecation messages are turned into compilation errors.)